// @samvera/clover-iiif/dist/scroll/index.mjs
import r, { createContext as U, useReducer as re, useContext as S, useState as b, useRef as P, useEffect as x } from "react";
import { Vault as L } from "@iiif/vault";
import ne from "sanitize-html";
import { createStitches as oe } from "@stitches/react";
import "hls.js";
import X from "openseadragon";
import "@iiif/vault-helpers";
import { ErrorBoundary as ie } from "react-error-boundary";
import { v4 as G } from "uuid";
import ae from "@nulib/use-markdown";
import le from "flexsearch";
const A = {
  annotations: [],
  manifest: void 0,
  searchString: void 0,
  options: {
    offset: 0
  },
  vault: new L()
};
function se(e, t) {
  switch (t.type) {
    case "updateAnnotations":
      return {
        ...e,
        annotations: t.payload
      };
    case "updateSearchString":
      return {
        ...e,
        searchString: t.payload
      };
    default:
      return e;
  }
}
const w = U({
  dispatch: () => null,
  state: A
}), ce = (e) => {
  const { children: t, manifest: n } = e, l = {
    ...A.options,
    ...e.options
  }, [o, i] = re(se, A);
  return /* @__PURE__ */ r.createElement(
    w.Provider,
    {
      value: {
        state: { ...o, manifest: n, options: l },
        dispatch: i
      }
    },
    t
  );
}, de = (e, t = "none") => {
  if (!e)
    return null;
  if (typeof e == "string")
    return [e];
  if (!e[t]) {
    const n = Object.getOwnPropertyNames(e);
    if (n.length > 0)
      return e[n[0]];
  }
  return !e[t] || !Array.isArray(e[t]) ? null : e[t];
}, M = (e, t = "none", n = ", ") => {
  const l = de(e, t);
  return Array.isArray(l) ? l.join(`${n}`) : l;
};
function ue(e) {
  return { __html: me(e) };
}
function T(e, t) {
  const n = Object.keys(e).filter(
    (o) => t.includes(o) ? null : o
  ), l = new Object();
  return n.forEach((o) => {
    l[o] = e[o];
  }), l;
}
function me(e) {
  return ne(e, {
    allowedAttributes: {
      a: ["href"],
      img: ["alt", "src", "height", "width"]
    },
    allowedSchemes: ["http", "https", "mailto"],
    allowedTags: [
      "a",
      "b",
      "br",
      "i",
      "img",
      "p",
      "small",
      "span",
      "sub",
      "sup"
    ]
  });
}
const k = 209, pe = {
  colors: {
    /*
     * Black and dark grays in a light theme.
     * Must contrast to 4.5 or greater with `secondary`.
     */
    primary: "#1a1d1e",
    primaryMuted: "#26292b",
    primaryAlt: "#151718",
    /*
     * Key brand color(s).
     * Must contrast to 4.5 or greater with `secondary`.
     */
    accent: `hsl(${k} 100% 38.2%)`,
    accentMuted: `hsl(${k} 80% 61.8%)`,
    accentAlt: `hsl(${k} 80% 30%)`,
    /*
     * White and light grays in a light theme.
     * Must contrast to 4.5 or greater with `primary` and  `accent`.
     */
    secondary: "#FFFFFF",
    secondaryMuted: "#e6e8eb",
    secondaryAlt: "#c1c8cd"
  },
  fontSizes: {
    1: "12px",
    2: "13px",
    3: "15px",
    4: "17px",
    5: "19px",
    6: "21px",
    7: "27px",
    8: "35px",
    9: "59px"
  },
  lineHeights: {
    1: "12px",
    2: "13px",
    3: "15px",
    4: "17px",
    5: "19px",
    6: "21px",
    7: "27px",
    8: "35px",
    9: "59px"
  },
  sizes: {
    1: "5px",
    2: "10px",
    3: "15px",
    4: "20px",
    5: "25px",
    6: "35px",
    7: "45px",
    8: "65px",
    9: "80px"
  },
  space: {
    1: "5px",
    2: "10px",
    3: "15px",
    4: "20px",
    5: "25px",
    6: "35px",
    7: "45px",
    8: "65px",
    9: "80px"
  },
  radii: {
    1: "4px",
    2: "6px",
    3: "8px",
    4: "12px",
    round: "50%",
    pill: "9999px"
  },
  transitions: {
    all: "all 300ms cubic-bezier(0.16, 1, 0.3, 1)"
  },
  zIndices: {
    1: "100",
    2: "200",
    3: "300",
    4: "400",
    max: "999"
  }
}, he = {
  xxs: "(max-width: 349px)",
  xs: "(max-width: 575px)",
  sm: "(max-width: 767px)",
  md: "(max-width: 991px)",
  lg: "(max-width: 90rem)",
  xl: "(min-width: calc(90rem + 1px))"
}, { styled: c, css: Nt, keyframes: Zt, createTheme: Ut } = oe({
  theme: pe,
  media: he
}), ge = c("span", {}), F = (e) => {
  const { as: t, label: n } = e, o = T(e, ["as", "label"]);
  return /* @__PURE__ */ r.createElement(ge, { as: t, ...o }, M(n, o.lang));
};
c("img", { objectFit: "cover" });
c("a", {});
const fe = {
  delimiter: ", "
}, R = U(void 0), xe = () => {
  const e = S(R);
  if (e === void 0)
    throw new Error(
      "usePrimitivesContext must be used with a PrimitivesProvider"
    );
  return e;
}, ve = ({
  children: e,
  initialState: t = fe
}) => {
  const n = we(t, "delimiter");
  return /* @__PURE__ */ r.createElement(R.Provider, { value: { delimiter: n } }, e);
}, we = (e, t) => Object.hasOwn(e, t) ? e[t].toString() : void 0, ye = c("span", {}), O = (e) => {
  const { as: t, markup: n } = e, { delimiter: l } = xe();
  if (!n)
    return /* @__PURE__ */ r.createElement(r.Fragment, null);
  const i = T(e, ["as", "markup"]), s = ue(
    M(n, i.lang, l)
  );
  return /* @__PURE__ */ r.createElement(ye, { as: t, ...i, dangerouslySetInnerHTML: s });
}, Se = (e) => r.useContext(R) ? /* @__PURE__ */ r.createElement(O, { ...e }) : /* @__PURE__ */ r.createElement(ve, null, /* @__PURE__ */ r.createElement(O, { ...e }));
c("span", {});
c("dl", {});
c("li", {});
c("ul", {});
c("li", {});
c("ul", {});
c("dl", {});
c("li", {});
c("ul", {});
const Ee = (e) => {
  const { as: t, summary: n } = e, o = T(e, ["as", "customValueDelimiter", "summary"]);
  return /* @__PURE__ */ r.createElement(Se, { as: t, markup: n, ...o });
}, _ = c("div", {
  display: "flex",
  flexDirection: "column",
  justifyContent: "space-between"
}), be = c("aside", {
  margin: "0",
  padding: "0",
  position: "relative",
  zIndex: 2,
  flexGrow: "1",
  flexShrink: "0",
  [`& ${_}`]: {
    position: "absolute",
    width: "50%",
    top: 0
  },
  "&.anchor": {
    [`& ${_}`]: {
      position: "fixed",
      width: "50%"
    }
  }
}), Ie = c("header", {
  display: "flex",
  justifyContent: "space-between",
  fontSize: "1",
  flexGrow: "1",
  flexShrink: "0",
  marginBottom: "1.618rem",
  strong: {
    fontWeight: "400",
    fontSize: "1.33rem"
  }
}), Ce = c("section", {
  margin: "0",
  gap: "1rem"
}), ke = ({ label: e }) => /* @__PURE__ */ r.createElement(Ie, null, /* @__PURE__ */ r.createElement("strong", null, /* @__PURE__ */ r.createElement(F, { label: e }))), $e = c("article", {
  transition: "all 0.382s ease-in-out",
  display: "flex",
  flexDirection: "row",
  flexWrap: "nowrap",
  gap: "2.618rem"
}), q = c("div", {
  transition: "$all",
  width: "50%",
  opacity: 0,
  transform: "translateX(2.618rem)",
  zIndex: -1
}), ze = c("div", {
  width: "50%",
  display: "flex",
  flexDirection: "column",
  justifyContent: "flex-start"
}), Le = c("hr", {
  margin: "0",
  borderColor: "transparent",
  height: "1.618rem",
  position: "relative",
  zIndex: 0,
  display: "flex",
  justifyContent: "flex-end",
  marginTop: "2.618rem",
  "&::before": {
    content: "attr(aria-label)",
    position: "absolute",
    right: "1.618rem",
    bottom: "0",
    zIndex: 1,
    display: "flex",
    fontSize: "0.7222rem",
    fontWeight: "400",
    lineHeight: "1rem",
    background: "inherit",
    opacity: 0.7
  },
  "&::after": {
    content: "",
    width: "100%",
    position: "absolute",
    zIndex: 0,
    height: "1px",
    background: "#6662"
  }
}), Ae = c("div", {
  position: "relative",
  zIndex: "1",
  display: "flex",
  flexDirection: "column",
  gap: "2.618rem",
  "&[data-figures-visible='true']": {
    [`& ${q}`]: {
      opacity: 1,
      zIndex: 0,
      transform: "translateX(0)"
    }
  }
}), _e = ({
  canvas: e,
  canvasInfo: { current: t, total: n }
}) => /* @__PURE__ */ r.createElement("figcaption", null, /* @__PURE__ */ r.createElement("em", null, t, " / ", n), (e == null ? void 0 : e.label) && /* @__PURE__ */ r.createElement(F, { label: e == null ? void 0 : e.label }), (e == null ? void 0 : e.summary) && /* @__PURE__ */ r.createElement(Ee, { summary: e == null ? void 0 : e.summary, as: "p" }));
let y = window.OpenSeadragon;
if (!y && (y = X, !y))
  throw new Error("OpenSeadragon is missing.");
const H = "http://www.w3.org/2000/svg";
y.Viewer && (y.Viewer.prototype.svgOverlay = function() {
  return this._svgOverlayInfo ? this._svgOverlayInfo : (this._svgOverlayInfo = new K(this), this._svgOverlayInfo);
});
const K = function(e) {
  const t = this;
  this._viewer = e, this._containerWidth = 0, this._containerHeight = 0, this._svg = document.createElementNS(H, "svg"), this._svg.style.position = "absolute", this._svg.style.left = 0, this._svg.style.top = 0, this._svg.style.width = "100%", this._svg.style.height = "100%", this._viewer.canvas.appendChild(this._svg), this._node = document.createElementNS(H, "g"), this._svg.appendChild(this._node), this._viewer.addHandler("animation", function() {
    t.resize();
  }), this._viewer.addHandler("open", function() {
    t.resize();
  }), this._viewer.addHandler("rotate", function() {
    t.resize();
  }), this._viewer.addHandler("flip", function() {
    t.resize();
  }), this._viewer.addHandler("resize", function() {
    t.resize();
  }), this.resize();
};
K.prototype = {
  // ----------
  node: function() {
    return this._node;
  },
  // ----------
  resize: function() {
    this._containerWidth !== this._viewer.container.clientWidth && (this._containerWidth = this._viewer.container.clientWidth, this._svg.setAttribute("width", this._containerWidth)), this._containerHeight !== this._viewer.container.clientHeight && (this._containerHeight = this._viewer.container.clientHeight, this._svg.setAttribute("height", this._containerHeight));
    const e = this._viewer.viewport.pixelFromPoint(new y.Point(0, 0), !0), t = this._viewer.viewport.getZoom(!0), n = this._viewer.viewport.getRotation(), l = this._viewer.viewport.getFlip(), o = this._viewer.viewport._containerInnerSize.x;
    let i = o * t;
    const s = i;
    l && (i = -i, e.x = -e.x + o), this._node.setAttribute(
      "transform",
      "translate(" + e.x + "," + e.y + ") scale(" + i + "," + s + ") rotate(" + n + ")"
    );
  },
  // ----------
  onClick: function(e, t) {
    new y.MouseTracker({
      element: e,
      clickHandler: t
    }).setTracking(!0);
  }
};
const Be = (e) => fetch(`${e.replace(/\/$/, "")}/info.json`).then((t) => t.json()).then((t) => t).catch((t) => {
  console.error(
    `The IIIF tilesource ${e.replace(
      /\/$/,
      ""
    )}/info.json failed to load: ${t}`
  );
}), Pe = (e) => {
  let t, n;
  if (Array.isArray(e) && (t = e[0], t)) {
    let l;
    "@id" in t ? l = t["@id"] : l = t.id, n = l;
  }
  return n;
};
var I = /* @__PURE__ */ ((e) => (e.TiledImage = "tiledImage", e.SimpleImage = "simpleImage", e))(I || {});
const Me = (e) => {
  const t = Array.isArray(e == null ? void 0 : e.service) && (e == null ? void 0 : e.service.length) > 0, n = t ? Pe(e == null ? void 0 : e.service) : e == null ? void 0 : e.id, l = t ? I.TiledImage : I.SimpleImage;
  return {
    uri: n,
    imageType: l
  };
}, Te = (e, t) => {
  const n = t ? I.TiledImage : I.SimpleImage;
  return {
    uri: e,
    imageType: n
  };
}, Fe = c("div", {
  display: "flex",
  flexDirection: "column",
  alignItems: "center"
}), Re = c("p", {
  fontWeight: "bold",
  fontSize: "x-large"
}), Oe = c("span", {
  fontSize: "medium"
}), He = ({ error: e }) => {
  const { message: t } = e;
  return /* @__PURE__ */ r.createElement(Fe, { role: "alert" }, /* @__PURE__ */ r.createElement(Re, { "data-testid": "headline" }, "Something went wrong"), t && /* @__PURE__ */ r.createElement(Oe, null, `Error message: ${t}`, " "));
}, B = c("div", {
  position: "absolute !important",
  zIndex: "1",
  top: "1rem",
  left: "1rem",
  width: "161.8px",
  height: "100px",
  backgroundColor: "#000D",
  boxShadow: "5px 5px 5px #0002",
  borderRadius: "3px",
  ".displayregion": {
    border: " 3px solid $accent !important",
    boxShadow: "0 0 3px #0006"
  },
  "@sm": {
    width: "123px",
    height: "76px"
  },
  "@xs": {
    width: "100px",
    height: "61.8px"
  }
}), We = c("div", {
  position: "relative",
  width: "100%",
  height: "100%",
  zIndex: "0"
}), Ve = c("div", {
  width: "100%",
  height: "100%",
  maxHeight: "100vh",
  background: "transparent",
  backgroundSize: "contain",
  color: "white",
  position: "relative",
  zIndex: "0",
  overflow: "hidden",
  variants: {
    hasNavigator: {
      true: {
        [`${B}`]: {
          display: "block"
        }
      },
      false: {
        [`${B}`]: {
          display: "none"
        }
      }
    }
  }
}), De = c("button", {
  display: "flex",
  height: "2rem",
  width: "2rem",
  borderRadius: "2rem",
  padding: "0",
  margin: "0",
  fontFamily: "inherit",
  background: "none",
  backgroundColor: "$primary",
  border: "none",
  color: "white",
  cursor: "pointer",
  marginLeft: "0.618rem",
  filter: "drop-shadow(2px 2px 5px #0003)",
  transition: "$all",
  boxSizing: "content-box !important",
  "&:first-child": {
    marginLeft: "0"
  },
  "@xs": {
    marginBottom: "0.618rem",
    marginLeft: "0",
    "&:last-child": {
      marginBottom: "0"
    }
  },
  svg: {
    height: "60%",
    width: "60%",
    padding: "20%",
    fill: "$secondary",
    stroke: "$secondary",
    filter: "drop-shadow(2px 2px 5px #0003)",
    transition: "$all",
    boxSizing: "inherit"
  },
  "&:hover, &:focus": {
    backgroundColor: "$accent"
  },
  "&[data-button=rotate-right]": {
    "&:hover, &:focus": {
      svg: {
        rotate: "45deg"
      }
    }
  },
  "&[data-button=rotate-left]": {
    transform: "scaleX(-1)",
    "&:hover, &:focus": {
      svg: {
        rotate: "45deg"
      }
    }
  },
  "&[data-button=reset]": {
    "&:hover, &:focus": {
      svg: {
        rotate: "-15deg"
      }
    }
  }
}), E = ({ className: e, id: t, label: n, children: l }) => {
  const o = n.toLowerCase().replace(/\s/g, "-");
  return /* @__PURE__ */ r.createElement(
    De,
    {
      id: t,
      className: e,
      "data-testid": "openseadragon-button",
      "data-button": o
    },
    /* @__PURE__ */ r.createElement(
      "svg",
      {
        xmlns: "http://www.w3.org/2000/svg",
        "aria-labelledby": `${t}-svg-title`,
        "data-testid": "openseadragon-button-svg",
        focusable: "false",
        viewBox: "0 0 512 512",
        role: "img"
      },
      /* @__PURE__ */ r.createElement("title", { id: `${t}-svg-title` }, n),
      l
    )
  );
}, je = c("div", {
  position: "absolute",
  zIndex: "1",
  top: "1rem",
  right: "1rem",
  display: "flex",
  "@xs": {
    flexDirection: "column",
    zIndex: "2"
  },
  variants: {
    hasPlaceholder: {
      true: {
        right: "3.618rem",
        "@xs": {
          top: "3.618rem",
          right: "1rem"
        }
      },
      false: {
        right: "1rem",
        "@xs": {
          top: "1rem",
          right: "1rem"
        }
      }
    }
  }
}), Ne = {
  behavior: "smooth",
  block: "center"
}, v = {
  annotationOverlays: {
    backgroundColor: "#6666ff",
    borderColor: "#000099",
    borderType: "solid",
    borderWidth: "1px",
    opacity: "0.5",
    renderOverlays: !0,
    zoomLevel: 2
  },
  background: "transparent",
  canvasBackgroundColor: "#6662",
  canvasHeight: "500px",
  contentSearch: {
    searchResultsLimit: 20,
    overlays: {
      backgroundColor: "#ff6666",
      borderColor: "#990000",
      borderType: "solid",
      borderWidth: "1px",
      opacity: "0.5",
      renderOverlays: !0,
      zoomLevel: 4
    }
  },
  ignoreCaptionLabels: [],
  informationPanel: {
    vtt: {
      autoScroll: {
        enabled: !0,
        settings: Ne
      }
    },
    open: !0,
    renderAbout: !0,
    renderSupplementing: !0,
    renderToggle: !0,
    renderAnnotation: !0,
    renderContentSearch: !0
  },
  openSeadragon: {},
  requestHeaders: { "Content-Type": "application/json" },
  showDownload: !0,
  showIIIFBadge: !0,
  showTitle: !0,
  withCredentials: !1,
  localeText: {
    contentSearch: {
      tabLabel: "Search Results",
      formPlaceholder: "Enter search words",
      noSearchResults: "No search results",
      loading: "Loading...",
      moreResults: "more results"
    }
  }
};
function Ze(e) {
  let t = {
    ...v.informationPanel.vtt.autoScroll
  };
  return typeof e == "object" && (t = "enabled" in e ? e : { enabled: !0, settings: e }), e === !1 && (t.enabled = !1), Ue(t.settings), t;
}
function Ue({ behavior: e, block: t }) {
  const n = ["auto", "instant", "smooth"], l = ["center", "end", "nearest", "start"];
  if (!n.includes(e))
    throw TypeError(`'${e}' not in ${n.join(" | ")}`);
  if (!l.includes(t))
    throw TypeError(`'${t}' not in ${l.join(" | ")}`);
}
var j, N;
const Xe = Ze(
  (N = (j = v == null ? void 0 : v.informationPanel) == null ? void 0 : j.vtt) == null ? void 0 : N.autoScroll
);
var Z;
const Y = {
  activeCanvas: "",
  activeManifest: "",
  OSDImageLoaded: !1,
  collection: {},
  configOptions: v,
  customDisplays: [],
  plugins: [],
  isAutoScrollEnabled: Xe.enabled,
  isAutoScrolling: !1,
  isInformationOpen: (Z = v == null ? void 0 : v.informationPanel) == null ? void 0 : Z.open,
  isLoaded: !1,
  isUserScrolling: void 0,
  vault: new L(),
  contentSearchVault: new L(),
  openSeadragonViewer: null,
  viewerId: G()
}, Ge = r.createContext(Y), qe = r.createContext(Y);
function W() {
  const e = r.useContext(Ge);
  if (e === void 0)
    throw new Error("useViewerState must be used within a ViewerProvider");
  return e;
}
function J() {
  const e = r.useContext(qe);
  if (e === void 0)
    throw new Error("useViewerDispatch must be used within a ViewerProvider");
  return e;
}
const Ke = () => /* @__PURE__ */ r.createElement(
  "path",
  {
    strokeLinecap: "round",
    strokeMiterlimit: "10",
    strokeWidth: "45",
    d: "M256 112v288M400 256H112"
  }
), Ye = () => /* @__PURE__ */ r.createElement(
  "path",
  {
    strokeLinecap: "round",
    strokeMiterlimit: "10",
    strokeWidth: "45",
    d: "M400 256H112"
  }
), Je = () => /* @__PURE__ */ r.createElement(
  "path",
  {
    fill: "none",
    stroke: "currentColor",
    strokeLinecap: "round",
    strokeLinejoin: "round",
    strokeWidth: "32",
    d: "M432 320v112H320M421.8 421.77L304 304M80 192V80h112M90.2 90.23L208 208M320 80h112v112M421.77 90.2L304 208M192 432H80V320M90.23 421.8L208 304"
  }
), Qe = () => /* @__PURE__ */ r.createElement("path", { d: "M448 440a16 16 0 01-12.61-6.15c-22.86-29.27-44.07-51.86-73.32-67C335 352.88 301 345.59 256 344.23V424a16 16 0 01-27 11.57l-176-168a16 16 0 010-23.14l176-168A16 16 0 01256 88v80.36c74.14 3.41 129.38 30.91 164.35 81.87C449.32 292.44 464 350.9 464 424a16 16 0 01-16 16z" }), V = () => /* @__PURE__ */ r.createElement(r.Fragment, null, /* @__PURE__ */ r.createElement(
  "path",
  {
    fill: "none",
    strokeLinecap: "round",
    strokeMiterlimit: "10",
    strokeWidth: "45",
    d: "M400 148l-21.12-24.57A191.43 191.43 0 00240 64C134 64 48 150 48 256s86 192 192 192a192.09 192.09 0 00181.07-128"
  }
), /* @__PURE__ */ r.createElement("path", { d: "M464 97.42V208a16 16 0 01-16 16H337.42c-14.26 0-21.4-17.23-11.32-27.31L436.69 86.1C446.77 76 464 83.16 464 97.42z" })), et = ({
  _cloverViewerHasPlaceholder: e,
  config: t
}) => {
  const n = W(), { activeCanvas: l, plugins: o, vault: i } = n, s = i.get({
    id: l,
    type: "Canvas"
  });
  function d() {
    return o.filter((a) => {
      var u;
      return (u = a.imageViewer) == null ? void 0 : u.controls;
    }).map((a, u) => {
      var p, g, h, f;
      const m = (g = (p = a.imageViewer) == null ? void 0 : p.controls) == null ? void 0 : g.component;
      return /* @__PURE__ */ r.createElement(
        m,
        {
          key: u,
          ...(f = (h = a == null ? void 0 : a.imageViewer) == null ? void 0 : h.controls) == null ? void 0 : f.componentProps,
          canvas: s,
          useViewerDispatch: J,
          useViewerState: W
        }
      );
    });
  }
  return /* @__PURE__ */ r.createElement(
    je,
    {
      "data-testid": "clover-iiif-image-openseadragon-controls",
      hasPlaceholder: e
    },
    t.showZoomControl && /* @__PURE__ */ r.createElement(r.Fragment, null, /* @__PURE__ */ r.createElement(E, { id: t.zoomInButton, label: "zoom in" }, /* @__PURE__ */ r.createElement(Ke, null)), /* @__PURE__ */ r.createElement(E, { id: t.zoomOutButton, label: "zoom out" }, /* @__PURE__ */ r.createElement(Ye, null))),
    t.showFullPageControl && /* @__PURE__ */ r.createElement(E, { id: t.fullPageButton, label: "full page" }, /* @__PURE__ */ r.createElement(Je, null)),
    t.showRotationControl && /* @__PURE__ */ r.createElement(r.Fragment, null, /* @__PURE__ */ r.createElement(E, { id: t.rotateRightButton, label: "rotate right" }, /* @__PURE__ */ r.createElement(V, null)), /* @__PURE__ */ r.createElement(E, { id: t.rotateLeftButton, label: "rotate left" }, /* @__PURE__ */ r.createElement(V, null))),
    t.showHomeControl && /* @__PURE__ */ r.createElement(E, { id: t.homeButton, label: "reset" }, /* @__PURE__ */ r.createElement(Qe, null)),
    d()
  );
}, tt = ({
  ariaLabel: e,
  config: t,
  uri: n,
  _cloverViewerHasPlaceholder: l,
  imageType: o,
  openSeadragonCallback: i
}) => {
  const [s, d] = b(), [a, u] = b(), m = J(), p = P(!1);
  return x(() => (p.current || (p.current = !0, a || u(X(t))), () => a == null ? void 0 : a.destroy()), []), x(() => {
    a && i && i(a);
  }, [a, i]), x(() => {
    a && n !== s && (a == null || a.forceRedraw(), d(n));
  }, [a, s, n]), x(() => {
    if (s && a)
      switch (o) {
        case "simpleImage":
          a == null || a.addSimpleImage({
            url: s
          });
          break;
        case "tiledImage":
          Be(s).then((g) => {
            try {
              if (!g)
                throw new Error(`No tile source found for ${s}`);
              a == null || a.addTiledImage({
                tileSource: g,
                success: () => {
                  typeof m == "function" && m({
                    type: "updateOSDImageLoaded",
                    OSDImageLoaded: !0
                  });
                }
              });
            } catch (h) {
              console.error(h);
            }
          });
          break;
        default:
          a == null || a.close(), console.warn(
            `Unable to render ${s} in OpenSeadragon as type: "${o}"`
          );
          break;
      }
  }, [o, s]), /* @__PURE__ */ r.createElement(
    Ve,
    {
      className: "clover-iiif-image-openseadragon",
      "data-testid": "clover-iiif-image-openseadragon",
      "data-openseadragon-instance": t.id,
      hasNavigator: t.showNavigator
    },
    /* @__PURE__ */ r.createElement(
      et,
      {
        _cloverViewerHasPlaceholder: l,
        config: t
      }
    ),
    t.showNavigator && /* @__PURE__ */ r.createElement(
      B,
      {
        id: t.navigatorId,
        "data-testid": "clover-iiif-image-openseadragon-navigator"
      }
    ),
    /* @__PURE__ */ r.createElement(
      We,
      {
        id: t.id,
        "data-testid": "clover-iiif-image-openseadragon-viewport",
        role: "img",
        ...e && { "aria-label": e }
      }
    )
  );
};
function rt(e) {
  return {
    id: `openseadragon-${e}`,
    navigatorId: `openseadragon-navigator-${e}`,
    loadTilesWithAjax: !0,
    fullPageButton: `fullPage-${e}`,
    homeButton: `reset-${e}`,
    rotateLeftButton: `rotateLeft-${e}`,
    rotateRightButton: `rotateRight-${e}`,
    zoomInButton: `zoomIn-${e}`,
    zoomOutButton: `zoomOut-${e}`,
    showNavigator: !0,
    showFullPageControl: !0,
    showHomeControl: !0,
    showRotationControl: !0,
    showZoomControl: !0,
    navigatorBorderColor: "transparent",
    gestureSettingsMouse: {
      clickToZoom: !0,
      dblClickToZoom: !0,
      pinchToZoom: !0,
      scrollToZoom: !1
    }
  };
}
const nt = ({
  _cloverViewerHasPlaceholder: e = !1,
  body: t,
  instanceId: n,
  isTiledImage: l = !1,
  label: o,
  src: i = "",
  openSeadragonCallback: s,
  openSeadragonConfig: d = {}
}) => {
  const a = n || G(), u = typeof o == "string" ? o : M(o), m = {
    ...rt(a),
    ...d
  }, { imageType: p, uri: g } = t ? Me(t) : Te(i, l);
  return g ? /* @__PURE__ */ r.createElement(ie, { FallbackComponent: He }, /* @__PURE__ */ r.createElement(
    tt,
    {
      _cloverViewerHasPlaceholder: e,
      ariaLabel: u,
      config: m,
      imageType: p,
      key: a,
      uri: g,
      openSeadragonCallback: s
    }
  )) : null;
}, ot = c("div", {
  width: "100%",
  height: "400px",
  background: "#6662",
  backgroundSize: "contain",
  color: "white",
  position: "relative",
  zIndex: "1",
  overflow: "hidden"
}), it = ({ body: e, label: t }) => /* @__PURE__ */ r.createElement(ot, { "data-testid": "scroll-figure-image" }, /* @__PURE__ */ r.createElement(
  nt,
  {
    body: e,
    openSeadragonConfig: { showNavigator: !1, showHomeControl: !1 },
    ...t && { label: t }
  }
)), at = c("figure", {
  figcaption: {
    display: "flex",
    flexDirection: "column",
    margin: "1.618rem 0 0",
    opacity: 0.9,
    em: {
      fontSize: "0.9em",
      fontStyle: "normal",
      opacity: 0.7
    }
  }
}), lt = (e, t, n, l) => {
  var s, d;
  const o = {
    canvas: void 0,
    accompanyingCanvas: void 0,
    annotationPage: void 0,
    annotations: []
  }, i = (a) => {
    if (a) {
      if (!a.body || !a.motivation) {
        console.error(
          "Invalid annotation after Hyperion parsing: missing either 'body' or 'motivation'",
          a
        );
        return;
      }
      let u = a.body;
      Array.isArray(u) && (u = u[0]);
      const m = e.get(u.id);
      if (!m)
        return;
      switch (n) {
        case "painting":
          return a.target === t.id && a.motivation && a.motivation[0] === "painting" && l.includes(m.type) && (a.body = m), !!a;
        case "supplementing":
          return;
        default:
          throw new Error("Invalid annotation motivation.");
      }
    }
  };
  if (o.canvas = e.get(t), o.canvas && (o.annotationPage = e.get(o.canvas.items[0]), o.accompanyingCanvas = (s = o.canvas) != null && s.accompanyingCanvas ? e.get((d = o.canvas) == null ? void 0 : d.accompanyingCanvas) : void 0), o.annotationPage) {
    const a = e.get(o.annotationPage.items).map((m) => ({
      body: e.get(m.body[0].id),
      motivation: m.motivation,
      type: "Annotation"
    })), u = [];
    a.forEach((m) => {
      m.body.type === "Choice" ? m.body.items.forEach(
        (p) => u.push({
          ...m,
          id: p.id,
          body: e.get(p.id)
        })
      ) : u.push(m);
    }), o.annotations = u.filter(i);
  }
  return o;
}, st = (e, t) => {
  const n = lt(
    e,
    { id: t, type: "Canvas" },
    "painting",
    ["Image", "Sound", "Video"]
  );
  if (n.annotations.length !== 0 && n.annotations && n.annotations)
    return n.annotations.map(
      (l) => l == null ? void 0 : l.body
    );
}, ct = ({ canvas: e, canvasInfo: t }) => {
  const { state: n } = S(w), { vault: l } = n, o = st(l, e.id);
  return o ? /* @__PURE__ */ r.createElement(at, null, o == null ? void 0 : o.map((i) => /* @__PURE__ */ r.createElement(it, { body: i, key: i == null ? void 0 : i.id, label: e == null ? void 0 : e.label })), /* @__PURE__ */ r.createElement(_e, { canvas: e, canvasInfo: t })) : null;
}, $ = "255, 197, 32", dt = c("div", {
  ul: {
    padding: "1rem"
  },
  li: {
    listStyleType: "disc",
    li: {
      listStyleType: "circle"
    }
  },
  "span.highlight": {
    position: "relative",
    fontWeight: "bold",
    "&::before": {
      top: "0",
      position: "absolute",
      display: "inline",
      content: "",
      width: "calc(100% + 4px)",
      height: "calc(100% + 2px) ",
      marginLeft: "-2px",
      borderRadius: "3px",
      border: `1px solid rgba(${$}, 0.2)`,
      borderBottom: `1px solid rgba(${$}, 0.618)`,
      boxShadow: "1px 1px 1px #6661"
    },
    "&::after": {
      left: "0",
      top: "0",
      position: "absolute",
      display: "inline",
      content: "",
      width: "calc(100% + 4px)",
      height: "calc(100% + 2px) ",
      marginLeft: "-2px",
      marginTop: "-1px",
      borderRadius: "3px",
      backgroundColor: `rgba(${$}, 0.2)`,
      zIndex: -1
    }
  }
}), ut = ({
  searchString: e = "",
  content: t,
  stringLength: n = 150
}) => {
  const l = t.toLowerCase().indexOf(e.toLowerCase());
  if (l !== -1) {
    let o = Math.max(0, l - n / 2), i = Math.min(
      t.length,
      l + e.length + n / 2
    );
    if (o > 0) {
      const s = t.lastIndexOf(" ", o + 1);
      o = s > 0 ? s + 1 : o;
    }
    if (i < t.length) {
      const s = t.indexOf(" ", i - 1);
      i = s > -1 ? s : i;
    }
    t = (o > 0 ? "... " : "") + t.substring(o, i) + (i < t.length ? " ..." : "");
  } else
    t = t.substring(0, n) + "...";
  return t;
}, Q = ({
  body: e,
  stringLength: t,
  type: n = "content"
}) => {
  const { state: l } = S(w), { searchString: o } = l;
  let i = String(e.value);
  n == "snippet" && (i = ut({
    searchString: o,
    content: i,
    stringLength: t
  }));
  let s;
  const d = ae(i);
  if (e.format === "text/plain") {
    const g = /\n/g;
    s = i == null ? void 0 : i.replace(g, "<br />");
  }
  if (e.format === "text/markdown" && (s = d == null ? void 0 : d.html), e.format === "text/html" && (s = i), o && s) {
    const g = new RegExp(`(${o})`, "gi");
    s = s == null ? void 0 : s.replace(
      g,
      (h) => `<span class="highlight">${h}</span>`
    );
  }
  const a = [e.id, n].join("-"), u = ["ar"].includes(String(e.language)), m = u ? "rtl" : "ltr", p = u ? "1.3em" : "1em";
  return s ? /* @__PURE__ */ r.createElement(
    dt,
    {
      dangerouslySetInnerHTML: { __html: s },
      "data-body-id": a,
      "data-testid": "scroll-item-body",
      id: a,
      dir: m,
      css: { fontSize: p }
    }
  ) : null;
}, mt = ({
  hasItemBreak: e,
  isLastItem: t,
  item: n,
  itemCount: l,
  itemNumber: o
}) => {
  var p;
  const { state: i } = r.useContext(w), { annotations: s, vault: d } = i, a = d == null ? void 0 : d.get(n), u = (p = s == null ? void 0 : s.filter((g) => g.target === n.id)) == null ? void 0 : p.map((g) => {
    var h;
    return (h = g == null ? void 0 : g.body) == null ? void 0 : h.map((f, C) => /* @__PURE__ */ r.createElement(
      Q,
      {
        body: f,
        key: C
      }
    ));
  }), m = {
    current: o,
    total: l
  };
  return /* @__PURE__ */ r.createElement(
    $e,
    {
      "data-page-break": e,
      "data-page-number": o,
      "data-last-item": t
    },
    /* @__PURE__ */ r.createElement(q, null, a && /* @__PURE__ */ r.createElement(ct, { canvas: a, canvasInfo: m })),
    /* @__PURE__ */ r.createElement(ze, null, (a == null ? void 0 : a.label) && /* @__PURE__ */ r.createElement("strong", null, /* @__PURE__ */ r.createElement(F, { label: a == null ? void 0 : a.label }), " ", `(${m.current} / ${m.total})`), /* @__PURE__ */ r.createElement("div", null, u || /* @__PURE__ */ r.createElement("p", null, "[Blank]")), e && /* @__PURE__ */ r.createElement(Le, { "aria-label": "Page Break" }))
  );
}, pt = r.memo(mt), z = "255, 197, 32", ht = c("span", {
  fontWeight: "700"
}), gt = c("div", {
  display: "flex",
  gap: "0.25rem"
}), ft = c("div", {
  fontSize: "0.9rem",
  padding: "1rem 0.618rem"
}), xt = c("div", {
  display: "flex",
  flexDirection: "column",
  gap: "0.5rem",
  hr: {
    margin: "0.618rem",
    border: "none",
    borderBottom: "1px solid #6662"
  },
  button: {
    backgroundColor: "#6660",
    opacity: "0.7",
    transition: "$all",
    padding: "0.5rem 0.618rem",
    fontSize: "0.9rem",
    lineHeight: "1.1rem",
    textAlign: "left",
    borderRadius: "6px",
    border: "1px solid #6662",
    display: "flex",
    flexDirection: "column",
    gap: "0.25rem",
    "&:hover": {
      opacity: "1",
      boxShadow: "5px 5px 13px #6662"
    },
    "&[data-result=true]": {
      backgroundColor: `rgba(${z}, 0.2)`,
      borderColor: `rgba(${z}, 0.2)`,
      opacity: "1",
      "&:hover": {
        backgroundColor: `rgba(${z}, 0.2)`
      }
    }
  }
}), ee = c("input", {
  margin: "0",
  padding: "0 1rem 0 2rem",
  background: "none",
  zIndex: "2",
  height: "2rem",
  marginLeft: "1rem",
  marginTop: "1rem",
  justifyContent: "center",
  display: "flex",
  alignItems: "center",
  fontSize: "1rem",
  borderRadius: "2rem",
  fontFamily: "inherit",
  backgroundColor: "$primary",
  border: "none",
  color: "$secondary",
  cursor: "text",
  filter: "drop-shadow(2px 2px 5px #0003)",
  transition: "$all",
  boxSizing: "content-box !important",
  flexGrow: "0",
  width: "100%",
  "&:placeholder": {
    color: "inherit"
  }
}), te = c("span", {
  position: "absolute",
  zIndex: "3",
  width: "2rem",
  height: "2rem",
  padding: "8px",
  marginTop: "1rem",
  marginLeft: "1rem",
  color: "$secondary",
  fill: "$secondary",
  stroke: "$secondary",
  transition: "$all",
  cursor: "text"
}), vt = c("button", {
  opacity: "1",
  display: "flex",
  alignItems: "center",
  borderRadius: "2rem",
  width: "2rem",
  height: "2rem",
  alignSelf: "center",
  marginTop: "1rem",
  gap: "0.35rem",
  backgroundColor: "$accent",
  fill: "$secondary",
  flexShrink: "0",
  svg: {
    padding: "6px",
    color: "inherit",
    fill: "inherit"
  },
  '&[aria-disabled="true"]': {
    opacity: "0",
    display: "none"
  }
}), wt = c("form", {
  display: "flex",
  justifyContent: "space-between",
  gap: "1rem",
  width: "100%",
  variants: {
    isPanelExpanded: {
      true: {
        [`${te}`]: {
          marginLeft: "0"
        },
        [`& ${ee}`]: {
          marginLeft: "0",
          backgroundColor: "$primary",
          width: "auto",
          flexGrow: "1"
          //
        }
      },
      false: {
        //
      }
    }
  }
}), yt = c("div", {}), D = ({
  annotation: e,
  isResult: t
}) => {
  var o, i;
  const n = [(o = e == null ? void 0 : e.body) == null ? void 0 : o.id, "content"].join("-"), l = (s) => {
    var d;
    s && ((d = document.getElementById(s)) == null || d.scrollIntoView({ behavior: "smooth" }));
  };
  return /* @__PURE__ */ r.createElement(
    "button",
    {
      "data-result": t ? "true" : "false",
      onClick: () => l(n),
      key: n
    },
    /* @__PURE__ */ r.createElement(gt, null, /* @__PURE__ */ r.createElement(ht, null, e.motivation), /* @__PURE__ */ r.createElement("span", null, (i = e == null ? void 0 : e.body) == null ? void 0 : i.language)),
    /* @__PURE__ */ r.createElement(
      Q,
      {
        body: e.body,
        stringLength: 144,
        type: "snippet"
      }
    )
  );
}, St = ({
  results: e
}) => /* @__PURE__ */ r.createElement(xt, null, e.found.map((t) => /* @__PURE__ */ r.createElement(
  D,
  {
    annotation: t,
    isResult: !0,
    key: t.id
  }
)), /* @__PURE__ */ r.createElement("hr", null), e.notFound.map((t) => /* @__PURE__ */ r.createElement(
  D,
  {
    annotation: t,
    key: t.id
  }
))), Et = {
  charset: "latin:extra, arabic:extra, cyrillic:extra, cjk:extra",
  optimize: !0,
  tokenize: "full",
  resolution: 9,
  document: {
    id: "id",
    index: "content"
  }
}, bt = () => {
  var a;
  const { state: e } = S(w), { annotations: t, searchString: n = "" } = e, l = new le.Document(Et), o = [];
  t == null || t.forEach((u) => {
    var m;
    (m = u == null ? void 0 : u.body) == null || m.forEach((p) => {
      var h;
      const g = (h = p == null ? void 0 : p.value) == null ? void 0 : h.replace(/\n/g, "");
      o.push(p == null ? void 0 : p.id), l.add({
        id: p == null ? void 0 : p.id,
        content: g
      });
    });
  });
  function i(u) {
    return u.map((m) => t == null ? void 0 : t.filter((p) => p.body.find((g) => g.id === m)).map((p) => {
      const g = p.body.findIndex((h) => h.id === m);
      return {
        ...p,
        body: p.body[g]
      };
    }).shift());
  }
  const s = l == null ? void 0 : l.search(n).reduce((u, m) => [.../* @__PURE__ */ new Set([...u, ...m.result])], []), d = {
    found: i(s),
    notFound: i(
      o.filter((u) => !s.includes(u))
    )
  };
  return /* @__PURE__ */ r.createElement(yt, null, n && /* @__PURE__ */ r.createElement(ft, null, (a = d.found) == null ? void 0 : a.length, " results for ", /* @__PURE__ */ r.createElement("strong", null, n)), /* @__PURE__ */ r.createElement(St, { results: d }));
}, It = () => /* @__PURE__ */ r.createElement("svg", { xmlns: "http://www.w3.org/2000/svg", viewBox: "0 0 512 512" }, /* @__PURE__ */ r.createElement("title", null, "Search"), /* @__PURE__ */ r.createElement("path", { d: "M456.69 421.39L362.6 327.3a173.81 173.81 0 0034.84-104.58C397.44 126.38 319.06 48 222.72 48S48 126.38 48 222.72s78.38 174.72 174.72 174.72A173.81 173.81 0 00327.3 362.6l94.09 94.09a25 25 0 0035.3-35.3zM97.92 222.72a124.8 124.8 0 11124.8 124.8 124.95 124.95 0 01-124.8-124.8z" })), Ct = () => /* @__PURE__ */ r.createElement("svg", { xmlns: "http://www.w3.org/2000/svg", viewBox: "0 0 512 512" }, /* @__PURE__ */ r.createElement("title", null, "Close"), /* @__PURE__ */ r.createElement("path", { d: "M289.94 256l95-95A24 24 0 00351 127l-95 95-95-95a24 24 0 00-34 34l95 95-95 95a24 24 0 1034 34l95-95 95 95a24 24 0 0034-34z" })), kt = ({
  togglePanel: e,
  isPanelExpanded: t
}) => {
  const { dispatch: n, state: l } = S(w), { searchString: o } = l, i = P(null), s = () => i.current === document.activeElement && e(!0), d = () => {
    var h;
    return (h = i == null ? void 0 : i.current) == null ? void 0 : h.focus();
  }, a = () => {
    var h;
    (h = i.current) == null || h.blur(), u(), e(!1), n({
      payload: "",
      type: "updateSearchString"
    });
  }, u = () => {
    i.current && (i.current.value = "");
  }, m = (h) => {
    h.preventDefault(), a();
  }, p = (h) => {
    h.key === "Escape" && (h.preventDefault(), a());
  };
  x(() => (document == null || document.addEventListener("keydown", p), () => {
    document == null || document.removeEventListener("keydown", p);
  }), []), x(() => {
    if (i != null && i.current)
      return i.current.addEventListener("focus", s), i.current.addEventListener("blur", s), () => {
        i.current && (i.current.removeEventListener("focus", s), i.current.removeEventListener("blur", s));
      };
  }, []);
  const g = (h) => {
    var f;
    n({
      payload: (f = h == null ? void 0 : h.target) == null ? void 0 : f.value,
      type: "updateSearchString"
    });
  };
  return /* @__PURE__ */ r.createElement(
    wt,
    {
      id: "scroll-search",
      autoComplete: "off",
      isPanelExpanded: t
    },
    /* @__PURE__ */ r.createElement(te, { onClick: d }, /* @__PURE__ */ r.createElement(It, null)),
    /* @__PURE__ */ r.createElement(
      ee,
      {
        ref: i,
        name: "clover-search",
        type: "text",
        placeholder: "Search...",
        defaultValue: o,
        onChange: g
      }
    ),
    /* @__PURE__ */ r.createElement(
      vt,
      {
        "aria-disabled": !t,
        "aria-label": "Close search panel",
        onClick: m,
        disabled: !t
      },
      /* @__PURE__ */ r.createElement(Ct, null)
    )
  );
}, $t = c("div", {
  display: "flex",
  flexDirection: "column",
  flexGrow: "1",
  flexShrink: "0",
  position: "relative",
  zIndex: "1",
  maxWidth: "100%",
  marginTop: "1rem",
  transition: "$all",
  variants: {
    isPanelExpanded: {
      true: {
        zIndex: 1,
        opacity: 1,
        transform: "translateX(0)"
      },
      false: {
        zIndex: -1,
        opacity: 0,
        transform: "translateX(-2.618rem)",
        transitionDelay: "0",
        transition: "none"
      }
    }
  }
}), zt = (e) => {
  const [t, n] = b({ top: 0, left: 0 });
  return x(() => {
    const l = () => {
      if (e.current) {
        const o = e.current.getBoundingClientRect();
        n({
          top: o.top,
          left: o.left
        });
      }
    };
    return window.addEventListener("scroll", l), window.addEventListener("resize", l), l(), () => {
      window.removeEventListener("scroll", l), window.removeEventListener("resize", l);
    };
  }, [e]), t;
}, Lt = ({
  isPanelExpanded: e,
  handlePanel: t
}) => {
  var g;
  const n = P(null), { state: l } = S(w), { options: o } = l, { offset: i } = o, { top: s } = zt(n), d = s ? s < i : !1, a = (g = n == null ? void 0 : n.current) == null ? void 0 : g.offsetWidth, u = a && a * 0.5, m = u ? u - 315 : 180, p = {
    top: d ? i : 0,
    width: `calc(${u}px - 1.318rem)`,
    maxWidth: e ? "100%" : `${m}px`,
    minWidth: "130px"
  };
  return /* @__PURE__ */ r.createElement(
    be,
    {
      ref: n,
      className: d ? "anchor" : "",
      "data-testid": "scroll-panel"
    },
    /* @__PURE__ */ r.createElement(_, { style: p }, /* @__PURE__ */ r.createElement(
      kt,
      {
        togglePanel: t,
        isPanelExpanded: e
      }
    ), /* @__PURE__ */ r.createElement(
      $t,
      {
        "data-testid": "scroll-panel-results",
        "data-panel-expanded": e,
        isPanelExpanded: e
      },
      e && /* @__PURE__ */ r.createElement(bt, null)
    ))
  );
}, At = ({ items: e }) => {
  const [t, n] = b(!1), l = (o) => n(o);
  return /* @__PURE__ */ r.createElement(r.Fragment, null, /* @__PURE__ */ r.createElement(
    Lt,
    {
      isPanelExpanded: t,
      handlePanel: l
    }
  ), /* @__PURE__ */ r.createElement(Ae, { "data-figures-visible": !t }, e.map((o, i) => {
    const s = i + 1, d = s === e.length;
    return /* @__PURE__ */ r.createElement(
      pt,
      {
        item: o,
        hasItemBreak: s < e.length,
        isLastItem: d,
        key: o.id,
        itemCount: e.length,
        itemNumber: s
      }
    );
  })));
}, _t = (e, t) => {
  const [n, l] = b([]);
  return x(() => {
    if (!t)
      return;
    const o = [];
    e == null || e.forEach((s) => {
      var a;
      const d = t.get(s);
      (a = d == null ? void 0 : d.annotations) == null || a.forEach(
        (u) => {
          var p;
          const m = t.get(u);
          (p = m == null ? void 0 : m.items) == null || p.forEach(
            (g) => {
              var f;
              const h = t.get(g);
              h && o.push({
                ...h,
                body: (f = h == null ? void 0 : h.body) == null ? void 0 : f.map((C) => t.get(C))
              });
            }
          );
        }
      );
    });
    const i = o.reduce(
      (s, d) => (s.some((a) => a.id === d.id) || s.push(d), s),
      []
    );
    l(i);
  }, [e, t]), n;
}, Bt = ({ iiifContent: e }) => {
  const [t, n] = b(), { state: l, dispatch: o } = S(w), { vault: i } = l, s = _t(t == null ? void 0 : t.items, i);
  return x(() => {
    i && i.load(e).then((d) => d && n(d)).catch(
      (d) => console.error(`Manifest ${e} failed to load: ${d}`)
    );
  }, [e, i]), x(() => {
    o({
      type: "updateAnnotations",
      payload: s
    });
  }, [s, o]), t ? /* @__PURE__ */ r.createElement(Ce, null, t.label && /* @__PURE__ */ r.createElement(ke, { label: t.label }), t.items && /* @__PURE__ */ r.createElement(At, { items: t.items })) : null;
}, Xt = ({
  iiifContent: e,
  options: t
}) => /* @__PURE__ */ r.createElement(ce, { options: t }, /* @__PURE__ */ r.createElement(Bt, { iiifContent: e }));
export {
  Xt as default
};
